From ca72b6081f8de433bb880df2ff31c25e3bd6af97 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:37:15 -0400
Subject: [virt] kvm/nvmx: Set success rflags when emulate VMXON/VMXOFF in nested virt

Message-id: <1406040016-3289-41-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86119
O-Subject: [RHEL7 PATCH v2 040/221] KVM: nVMX: Set success rflags when emulate VMXON/VMXOFF in nested virt
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Arthur Chunqi Li <yzt356@gmail.com>

Set rflags after successfully emulateing VMXON/VMXOFF in VMX.

Signed-off-by: Arthur Chunqi Li <yzt356@gmail.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit a25eb114d50d6e60ef586f2466c874581c31594c)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/vmx.c b/arch/x86/kvm/vmx.c
index b9fddf2..d87fbd4 100644
--- a/arch/x86/kvm/vmx.c
+++ b/arch/x86/kvm/vmx.c
@@ -5678,6 +5678,7 @@ static int handle_vmon(struct kvm_vcpu *vcpu)
  vmx->nested.vmxon = true;
 
  skip_emulated_instruction(vcpu);
+ nested_vmx_succeed(vcpu);
  return 1;
 }
 
@@ -5762,6 +5763,7 @@ static int handle_vmoff(struct kvm_vcpu *vcpu)
   return 1;
  free_nested(to_vmx(vcpu));
  skip_emulated_instruction(vcpu);
+ nested_vmx_succeed(vcpu);
  return 1;
 }
 
-- 
1.7.1